### PR TITLE
Graph: Adds onHorizontalRegionSelected

### LINKED ISSUE #18476: React Graph: Make onSelectionChanged work
**What would you like to be added**:
When solving #17302 we introduced the ability to select in the React Graph. That was only "hooked" up in the Explore part and not for GraphPanel. During this we should also explore the possibility make this prop more generic as suggested here: https://github.com/grafana/grafana/pull/18475#discussion_r312927518

**Why is this needed**:

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Closes #18476

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/graph2/GraphPanelController.tsx [hunk 1]
```diff
@@ -113,6 +116,13 @@ export class GraphPanelController extends React.Component<GraphPanelControllerPr
     });
   }
 
+  onHorizontalRegionSelected(from: number, to: number) {
+    getTimeSrv().setTime({
```
Comment (hugohaggmark) [inline public/app/plugins/panel/graph2/GraphPanelController.tsx]: This feels really dirty, is there a better way that is uncoupled from the Angular service?

Comment (torkelo) [reply public/app/plugins/panel/graph2/GraphPanelController.tsx]: yea, my thinking is to have a onChangeTimeRange on the PanelProps instead so panel plugins do not need to access TimeSrv

Comment (hugohaggmark) [reply public/app/plugins/panel/graph2/GraphPanelController.tsx]: Moved it to PanelChrome instead

File: packages/grafana-ui/src/types/panel.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ export interface PanelProps<T = any> {
   width: number;
   height: number;
   replaceVariables: InterpolateFunction;
+  onChangeTimeRange: (timeRange: AbsoluteTimeRange) => void;
```
Comment (torkelo) [inline packages/grafana-ui/src/types/panel.ts]: @ryantxu adding this callback to PanelProps, any thoughts? only thing I am not 100% sure about is if we should support the text form as well (the string relative date math expressions), or we can wait and add that later as a second parameter. Or if we should use TimeRange (that has from:/to in DateTime, and raw with the string versions)

**Review by torkelo — APPROVED** [review]

**Pull request merged**